pkg/resources/convertible.go

```diff
@@ -34,8 +39,16 @@ func ConvertibleResources() ([]ConvertibleResourceDef, error) {
 		tfRes := tfResFn()
 		cRes, ok := tfRes.(ConvertibleResourceDef)
 		if !ok {
-			err := fmt.Errorf("terraform resource %T does not implement ConvertibleResourceDef", tfRes)
-			return nil, err
+			not_convertible, ok := tfRes.(NotConvertibleResourceDef)
+			if !ok {
+				err := fmt.Errorf("terraform resource %T does not implement ConvertibleResourceDef nor NotConvertibleResourcedef", tfRes)
+				return nil, err
+			}
+			if !not_convertible.NotConvertible() {
+				err := fmt.Errorf("terraform resource %T must return `true` from `NotConvertible()`", tfRes)
+				return nil, err
+			}
+			continue
```

Comment:
We can get rid of some of the nail checks if we stop storing by index number and start appending.

Reply:
fixed